packages/grafana-ui/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { PureComponent, useContext, FC } from 'react';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { ThemeContext } from '../../themes/index';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '../../types/theme';
+import tinycolor from 'tinycolor2';
+import { ToggleButtonGroupProps, ToggleButtonProps, ToggleButtonState } from './types';
+import { increaseContrast } from '../../utils/colors';
+import { ButtonSize } from '../Button/AbstractButton';
+
+const ToggleButtonGroup: FC<ToggleButtonGroupProps> = ({ label, children, propsPriority }) => {
+  const styles = getStyles(useContext(ThemeContext));
+  return (
+    <div>
+      <span className={styles.label}>{label}</span>
+      {React.Children.map(children, (child: any, index) => {
+        // Set correct style depending on first/middle/last
+        const style = index === 0 ? styles.first : index === children.length - 1 ? styles.last : styles.middle;
+
+        return React.cloneElement(child, {
+          className: style,
+          propsPriority,
+        });
+      })}
+    </div>
+  );
+};
+
+ToggleButtonGroup.displayName = 'ToggleButtonGroup';
+
+export { ToggleButtonGroup };
+export class ToggleButton extends PureComponent<ToggleButtonProps, ToggleButtonState> {
+  constructor(props: ToggleButtonProps) {
+    super(props);
+    this.state = {
+      untouched: true,
+      selected: props.selected,
+      value: props.value,
+    };
+
+    this.handleBlur = this.handleBlur.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e: React.SyntheticEvent) {
+    e.stopPropagation();
+
+    this.setState(prevState => ({
+      untouched: false,
+      selected: this.props.propsPriority ? this.props.selected : !prevState.selected,
+    }));
+    if (this.props.onChange) {
+      this.props.onChange(this.state);
+    }
+  }
+
+  handleBlur() {
+    this.setState({
+      untouched: true,
+    });
+  }
+
+  componentDidUpdate(prevProps: ToggleButtonProps) {
+    //Override internal select toggle if props are updated
+    if (prevProps.selected !== this.props.selected) {
+      this.setState({
+        selected: this.props.selected,
```

Comment:
This is going to cause additional re-render. You can use https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Did this because I didn't want to "pollute" the state since it is passed onto `onChange`, but maybe that's an irrational fear? `getDerivedStateFromProps` works for me.